src/models/room.model.ts

```diff
@@ -38,6 +38,22 @@ class Room
             }
         });
     }
+
+    static updateById = (data: RoomData, callback: Callback) => {
+        sql.query(
+            'UPDATE rooms SET ? WHERE id = ?',
+            [{ title: data.title, body: data.body }, data.id],
+            (error, res) => {
+                if (error) {
+                    callback({ error, message: 'Mysql error' });
+                } else if (!res.affectedRows) {
+                    callback({ message: 'No room found' });
+                } else {
+                    callback(null, res[0]);
```

Comment:
```suggestion
                    callback(null, data);
```
`res[0]` is undefined apparently.